In [2]:
import pandas as pd
import numpy as np
import re

In [3]:
movies_df = pd.read_csv('output/disney_movies.csv')
movies_df.head()

movieId                            title  \
0      594  Snow White and the Seven Dwarfs   
1      596                        Pinocchio   
2     5990                        Pinocchio   
3    90264                        Pinocchio   
4   126693                        Pinocchio   

                                     genres  
0  Animation|Children|Drama|Fantasy|Musical  
1        Animation|Children|Fantasy|Musical  
2                   Children|Comedy|Fantasy  
3                  Children|Fantasy|Musical  
4                        Animation|Children

You can see above that Pinocchio appears 4 times with different movieId. What I would like to do is merge the df above without losing the movieId and genres. What this means is, you must find the duplicates, keep the movieId in a replacement_df (old_movieId, new_movieId). Then you search by length <len()> of genres, and keep the movieId for the one with longest "genres".

So for movies_df above, assuming there are 4 "Pinocchio", then you want to keep movieId = 596 because the length is the longest for genres.

That also means, your replacement_df will be like this: old_movieId, new_movieId:- [5990,596] ; [90264,596] ; [126693,596]. This replacement_df is important because you need to replace for the ratings_df below too. 

In [4]:
ratings_df = pd.read_csv('output/disney_ratings.csv')
ratings_df.head()

userId  movieId  rating   timestamp
0       1     6377     4.0  1147868469
1       3     6377     4.0  1439472644
2       4     6377     3.0  1573939584
3      12     6377     4.0  1167582434
4      13     6377     4.0  1238029123